Assets/MRTK/Providers/LeapMotion/LeapMotionDeviceManager.cs

```diff
@@ -0,0 +1,242 @@
+        /// <param name="priority">Service priority. Used to determine order of instantiation.</param>
+        /// <param name="profile">The service's configuration profile.</param>
+        public LeapMotionDeviceManager(
+            IMixedRealityInputSystem inputSystem,
+            string name = null,
+            uint priority = DefaultPriority,
+            BaseMixedRealityProfile profile = null) : base(inputSystem, name, priority, profile) { }
+
+
+#region IMixedRealityCapabilityCheck Implementation
+
+        /// <inheritdoc />
+        public bool CheckCapability(MixedRealityCapability capability)
+        {
+            // Leap Motion only supports Articulated Hands
+            return (capability == MixedRealityCapability.ArticulatedHand);
+        }
+
+
+#endregion IMixedRealityCapabilityCheck Implementation
+#if LEAPMOTIONCORE_PRESENT
+
+        /// <summary>
+        /// The profile that contains settings for the Leap Motion Device Manager input data provider.  This profile is nested under 
+        /// Input > Input Data Providers > Leap Motion Device Manager in the MixedRealityToolkit object in the hierarchy.
+        /// </summary>
+        public LeapMotionDeviceManagerProfile SettingsProfile => ConfigurationProfile as LeapMotionDeviceManagerProfile;
+
+        // This value can only be set in the profile, the default is LeapControllerOrientation.Headset
+        private LeapControllerOrientation leapControllerOrientation => SettingsProfile.LeapControllerOrientation;
+
+        /// <summary>
+        /// Adds an offset to the game object with LeapServiceProvider attached.  This offset is only applied if the leapControllerOrientation
+        /// is LeapControllerOrientation.Desk and is nessesary for the hand to appear in front of the main camera. If the leap controller is on the 
+        /// desk, the LeapServiceProvider is added to the scene instead of the LeapXRServiceProvider. The anchor point for the position of the leap hands is 
+        /// the position of the game object with the LeapServiceProvider attached.
+        /// </summary>
+        private Vector3 leapHandsOffset => SettingsProfile.LeapControllerOffset;
+
+        /// <summary>
+        /// If true, the leap motion controller connected and detected.
+        /// </summary>
+        public bool IsLeapConnected => leapServiceProvider.IsConnected();
+
+        /// <summary>
+        /// Dictionary to capture all active leap motion hands detected.
+        /// </summary>
+        private readonly Dictionary<Handedness, LeapMotionArticulatedHand> trackedHands = new Dictionary<Handedness, LeapMotionArticulatedHand>();
+
+        // The LeapServiceProvider is added to the scene at runtime in OnEnable 
+        public LeapServiceProvider leapServiceProvider { get; protected set;}
+
+        // The Leap attachment hands, used to determine which hand is currently tracked by leap
+        public AttachmentHands attachmentHands { get; protected set; }
```

Comment:
To fit [our naming conventions](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/CodingGuidelines.html#naming-conventions), `leapServiceProvider` and `attachmentHands` should be in PascalCase since they're properties.